packages/hadron-type-checker/index.d.ts

```diff
@@ -20,8 +20,8 @@ export type TypeCastMap = {
   Code: Code;
   Date: Date;
   Decimal128: Decimal128;
-  Double: Double;
-  Int32: Int32;
+  Double: Double | number;
+  Int32: Int32 | number;
```

Comment:
Ah, you know, yeah, you're totally right, that's not the right place for this change. The reason for this was that Document / Elements will accept a JavaScript number as value and type checker will return the type for those as int32 / double, so you can have a state of these classes where `el.currentValue` will not be an instance of bson type but the `currentType` will be (similar to what we would get in the app if we don't pass `promoteValues: false` to the driver) one of these and this type accommodates for that and guarantees that the code that will access this data will need to handle that, but then I guess this should be then adjusted on the `hadron-document` side, not in `type-checker`, maybe we should just always try to cast when the instance is created or something 🤔

Reply:
> maybe we should just always try to cast when the instance is created or something

Yeah, that sounds right to me – I don’t think this should happen with how we typically interact with BSON documents that we handle via hadron-document, but if we want to accept JS numbers here, we should probably convert them to a full BSON class instance so that the Element’s `value` property is always one of those.

Reply:
Done in f3c25025d1e78431266b20f2da8589737c45d68b